common/src/test/scala/com/nvidia/spark/rapids/ThreadFactoryBuilderTest.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids
+
+import java.util.concurrent.Executors
+
+import org.scalatest.FunSuite
+
+class ThreadFactoryBuilderTest extends FunSuite {
+
+  test("test thread factory builder") {
+    var pool = Executors.newFixedThreadPool(2,
+      new ThreadFactoryBuilder().setNameFormat("Thread %s").setDaemon(true).build())
+
+    pool.submit(new Runnable {
+      override def run(): Unit = {
+        assert(Thread.currentThread().isDaemon)
+        assert(Thread.currentThread().getName == "Thread 0")
+      }
+    })
```

Comment:
I don't think this does what we want.  The main thread, i.e: the one responsible for the test, will submit a bunch of tasks to executors and does _not_ wait for them to complete nor evaluate any `Future` result from their run.  In other words, the main thread could race through and "succeed" before the first submitted task is even run.  In addition, the tasks will throw exceptions if they assert, but these will not be seen by the main test thread.  Instead the executor service will catch these exceptions and hold onto them, waiting for some task to examine the `Future` that was returned by the `submit` so it can propagate the exception that was caught when someone tries to get the result from the `submit`.  However since that is thrown away, so its the exception -- it will be silently eaten.

For this test to be valid, the main thread minimally needs to wait for the submitted tasks by examining the `Future` results so any exceptions thrown by the worker threads will be propagated to the main thread and thus fail the test.